core/dbt/logger.py

```diff
@@ -265,31 +265,55 @@ def process(self, record):
         record.extra['node_count'] = self.node_count
 
 
-class NodeMetadata(logbook.Processor):
-    def __init__(self, node, node_index):
-        self.node = node
-        self.node_index = node_index
+class ModelMetadata(logbook.Processor):
+    def __init__(self, model, index):
+        self.model = model
+        self.index = index
         super().__init__()
 
+    def mapping_keys(self):
+        return []
+
+    def process_keys(self, record):
+        for attr, key in self.mapping_keys():
+            value = getattr(self.model, attr, None)
+            if value is not None:
+                record.extra[key] = value
+
     def process(self, record):
-        keys = [
+        self.process_keys(record)
+        record.extra['node_index'] = self.index
+
+
+class NodeMetadata(ModelMetadata):
```

Comment:
This looks backwards to me - do we not want `ModelMetadata` to inherit from `NodeMetadata`? Not super important, mostly just curious what the thinking is here.

Reply:
You're totally right, fixed